File: java/cours/src/session2/td/heritageMultiple/NatParInt.java
package session2.td.heritageMultiple;

import session1.td.Nat;

public class NatParInt extends IntPositif implements AlgebreNatParInt{

	public NatParInt(int rep) {
		super(rep);
	}
	
	@Override
	public Nat creerNatAvecValeur(int val) {
		return new NatParInt(val);
	}
	
	@Override
	public String toString() {
		return Integer.toString(this.val());
	}
	@Override
	public boolean equals(Object obj) {
		if(!(obj instanceof Nat))
			return false;
		Nat x = (Nat)obj;
		return this.val() == x.val();
	}
}

File: typescript/session2/heritageMultiple/NatParInt.ts
import {IntPositif} from "./IntPositif";
import {AlgebreNatParInt} from "./AlgebreNatParInt";
import {Nat} from "../../session1/td/Nat/Nat";

export class NatParInt extends IntPositif, AlgebreNatParInt{

    constructor(rep : number) {
        super(rep);
    }

    creerNatAvecValeur(val : number) : Nat{
        return new NatParInt(val);
    }

    toString() : string{
        return '' + this.getVal();
    }

    equals(obj : Object) : boolean {
        if(!(obj instanceof Nat))
            return false;
        let x = <Nat>obj;
        return this.getVal() == x.getVal();
    }
}
